numalogic/udfs/preprocess.py

```diff
@@ -99,9 +100,15 @@ def exec(self, keys: list[str], datum: Datum) -> Messages:
         _stream_conf = self.get_stream_conf(data_payload["config_id"])
         _conf = _stream_conf.ml_pipelines[data_payload["pipeline_id"]]
         raw_df, timestamps = get_df(data_payload=data_payload, stream_conf=_stream_conf)
-
-        # TODO: Add pipeline id to the ml metrics
+        try:
+            source = data_payload["metadata"]["numalogic_opex_tags"]["source"]
+        except Exception:
```

Comment:
Wont just catching KeyError work?

Reply:
We will otherwise have to check existence of keys: meatadata, numalogic_opex_tags and source.

Reply:
What do you think?